pysteps/blending/ens_kalman_filter_methods.py

```diff
@@ -0,0 +1,538 @@
+        X_bg: np.ndarray,
+        Y_obs: np.ndarray,
+        inflation_factor_bg: float,
+        inflation_factor_obs: float,
+        offset_bg: float,
+        offset_obs: float,
+        X_bg_lien: np.ndarray | None = None,
+        Y_obs_lien: np.ndarray | None = None,
+    ):
+        """
+        Compute the ensemble Kalman filter update step.
+
+        Parameters
+        ----------
+        X_bg: np.ndarray
+            Two-dimensional array of shape (n_ens, n_pc) containing the background
+            ensemble that corresponds to the Nowcast ensemble forecast.
+        Y_obs: np.ndarray
+            Two-dimensional array of shape (n_ens, n_pc) containing the observations
+            that correspond to the NWP ensemble forecast.
+        inflation_factor_bg: float
+            Inflation factor of the background ensemble covariance matrix.
+        inflation_factor_obs: float
+            Inflation factor of the observation covariance matrix.
+        offset_bg: float
+            Offset of the background ensemble covariance matrix.
+        offset_obs: float
+            Offset of the observation covariance matrix.
+
+        Other Parameters
+        ----------------
+        X_bg_lien: np.ndarray
+            Two-dimensional array of shape (n_ens, n_pc) containing the background
+            ensemble that consists only of grid boxes at which the Lien criterion is
+            satisfied.
+        Y_obs_lien: np.ndarray
+            Two-dimensional array of shape (n_ens, n_pc) containing the observations
+            that consists only of grid boxes at which the Lien criterion is satisfied.
+
+        Returns
+        -------
+        X_ana: np.ndarray
+            Two-dimensional array of shape (n_ens, n_pc) containing the updated
+            analysis matrix.
+        """
+
+        # If the masked background and observation arrays are given, compute the
+        # covariance matrices P and R only on these values...
+        if X_bg_lien is not None and Y_obs_lien is not None:
+
+            P = self.get_covariance_matrix(
+                X_bg_lien, inflation_factor=inflation_factor_bg, offset=offset_bg
+            )
+
+            R = self.get_covariance_matrix(
+                Y_obs_lien, inflation_factor=inflation_factor_obs, offset=offset_obs
+            )
+        # ...otherwise use the complete arrays.
+        else:
+
+            P = self.get_covariance_matrix(
+                X_bg, inflation_factor=inflation_factor_bg, offset=offset_bg
+            )
+
+            R = self.get_covariance_matrix(
+                Y_obs, inflation_factor=inflation_factor_obs, offset=offset_obs
+            )
+
+        # Estimate the Kalman gain
+        self.K = np.dot(P, np.linalg.inv(P + R))
+
+        # Update the background ensemble
+        X_ana = X_bg.T + np.dot(self.K, (Y_obs - X_bg).T)
+
+        return X_ana
+
+    def get_covariance_matrix(self, M: np.ndarray, inflation_factor: float, offset: float):
+        """
```

Comment:
In this function, can you refer to the equation(s) from Nerini et al. (2019) that are used?